src/ipinfo.rs

```diff
@@ -130,6 +146,17 @@ impl IpInfo {
         let mut details: HashMap<String, IpDetails> =
             serde_json::from_str(&raw_resp)?;
 
+        // Add country_name and EU status to response
+        for detail in details.to_owned() {
+            let mut_details = details.get_mut(&detail.0).unwrap();
+            let country = &mut_details.country;
+            if !country.is_empty() {
+                let country_name = countries.get(&mut_details.country).unwrap();
+                mut_details.countryname = Some(country_name.to_string());
+                mut_details.is_eu = Some(eu_countries.contains(country));
+            }
+        }
+
         // Update cache
         details.iter().for_each(|x| {
```

Comment:
If we're already iterating over here, we should probably just set the country_name and is_eu over here, and update the comments a bit so it makes more sense.

Reply:
we need to `own` the data for changing the values, and values get updated when the iteration is finished. I tried putting the values to the cache in the same loop where we assigned the values of country_name and is_eu but it's adding `country_name: None` and `is_eu: None` in the cache. That's why we have to go with this approach.

Reply:
I see. :rofl: